### ISSUE TITLE
Use of tagNamePreference breaks all @constructor tags

### ISSUE THREAD

**Issue by tschaub** [gajus/eslint-plugin-jsdoc#898]
## Expected behavior

I would like to use the `jsdoc/check-tag-names` with `tagNamePreference` in my settings and still use tags like `@constructor` even though they are named like `Object.prototype` properties.

## Actual behavior

```
Invalid JSDoc tag (preference). Replace "constructor" JSDoc tag with "function Object() { [native code] }"  jsdoc/check-tag-names
```

## ESLint Config

In `package.json`, here is my `eslintConfig`:
```json
{
  "plugins": [
    "jsdoc"
  ],
  "rules": {
    "jsdoc/check-tag-names": "error"
  },
  "settings": {
    "jsdoc": {
      "tagNamePreference": {
        "returns": "return"
      }
    }
  }
}
```

## ESLint sample

```js
/**
 * @constructor
 */
function Test() {
  this.works = false;
}
```

## Environment

- Node version: v18.4.0
- ESLint version: v8.21.0
- `eslint-plugin-jsdoc` version: v39.3.4

This is different than the issue described in #785, where the user was trying to provide a `tagNamePreference` for the `@constructor` tag.  In my case, I'm trying to provide a `tagNamePreference` for any tag (not only one that conflicts with `Object.prototype` names), and this makes it so I can no longer use the `@constructor` tag.
